packages/puppeteer-core/src/api/locators/Locator.ts

```diff
@@ -1,28 +1,51 @@
+import {
+  EMPTY,
+  Observable,
+  OperatorFunction,
+  catchError,
+  defaultIfEmpty,
+  defer,
+  filter,
+  first,
+  firstValueFrom,
+  from,
+  fromEvent,
+  identity,
+  ignoreElements,
+  map,
+  merge,
+  mergeMap,
+  noop,
+  pipe,
+  raceWith,
+  retry,
+  tap,
+  timer,
+} from '../../../third_party/rxjs/rxjs.js';
+import {TimeoutError} from '../../common/Errors.js';
 import {EventEmitter} from '../../common/EventEmitter.js';
-import {ClickOptions} from '../ElementHandle.js';
+import {HandleFor} from '../../common/types.js';
+import {debugError} from '../../common/util.js';
+import {BoundingBox, ClickOptions, ElementHandle} from '../ElementHandle.js';
 
 import {
+  Action,
+  AwaitedLocator,
   ExpectedLocator,
   Predicate,
   RaceLocator,
-  UnionLocatorOf,
-  type ActionCondition,
 } from './locators.js';
 
 /**
+ * For observables coming from promises, a delay is needed, otherwise RxJS will
+ * never yield in a permanent failure for a promise.
+ *
+ * We also don't want RxJS to do promise operations to often, so we bump the
+ * delay up to 100ms.
+ *
  * @internal
  */
-export interface LocatorContext<T> {
-  conditions?: Set<ActionCondition<T>>;
-}
-
-/**
- * @internal
- */
-export const LOCATOR_CONTEXTS = new WeakMap<
-  Locator<unknown>,
-  LocatorContext<never>
->();
+export const RETRY_DELAY = 100;
```

Comment:
Non-blocking: That looks like a little bit too long imo I would set it to 50ms, @OrKoN what do you think.

Reply:
it's been 100ms in the previous version. 100ms sounds good to me.